include/hss_connection.h

```diff
@@ -0,0 +1,131 @@
+/**
+ * @file hss_connection.h Abstract base class representing connection to an HSS.
+ *
+ * Copyright (C) Metaswitch Networks 2017
+ * If license terms are provided to you in a COPYING file in the root directory
+ * of the source code repository by which you are accessing this code, then
+ * the license outlined in that COPYING file applies to your use.
+ * Otherwise no rights are granted except for those provided to you by
+ * Metaswitch Networks in a separate written agreement.
+ */
+#ifndef HSS_CONNECTION_H__
+#define HSS_CONNECTION_H__
+
+#include <string>
+
+// This enum represents the various responses we can get from the HSS.
+// The DiameterHssConnection will map from combinations of Diameter result code
+// and experimental result code to one of these.
+// Not complete
+enum HssResultCode
+{
+    SUCCESS = 0,
+    SERVER_UNAVAILABLE = 1,
+    NOT_FOUND = 2
+};
+
+// Base class that represents a response from the HSS.
+// We always have an HssResultCode
+class HssResponse
+{
+public:
+  virtual ~HssResponse();
+
+  HssResponse(HssResultCode rc) : _result_code(rc) {};
+
+  HssResultCode get_result()
+  {
+    return _result_code;
+  }
+
+private:
+  HssResultCode _result_code;
+};
+
+// Subclasses of the HssResponse.
+// We will have 4 - one for each of the 4 requests we make to the HSS.
+// Each one holds the data that the handlers need to process the response.
+class MultimediaAuthResponse : public HssResponse
```

Comment:
This is confusing - MAR may mean MultimediaAuthRequest (in Diameter) or MultimediaAuthResponse (in this class). The Diameter equivalent is MultimediaAuthAnswer.

I think we either need different names which describe what they actually do (eg. GetAuthDetails, GetProfile, LookupAuthDetails, FindUser etc), or we need to follow the Diameter names exactly.